lifting-state-up/single-source-of-truth.md

```diff
@@ -0,0 +1,260 @@
+
+In Sofia's app, all components that read student data, should reference it from the same source.
+
+Right now, Sofia's app introduces student data in two places:
+
+- The `App` component, which holds student names and emails
+- The `Student` component, which manages student `isPresent` details
+
+### !callout-info
+
+## The Flip Side: Managing Multiple Sources of "Truth"
+
+Consider for a moment what it would be like if we _didn't_ strive for a single source of truth. As we continued to develop our app, we'd possibly ask the following questions:
+
+- If another component needs to read a student's `isPresent` state, where do they look for that data?
+  - For example, what if another component wanted to count how many students were absent that day?
+- If there are multiple sources of truth, in what situations do we update one of source of truth, and don't update the others?
+- What happens if our multiple sources of truth are out of sync with each other?
+
+<br/>
+
+These questions highlight problems that are challenging to work with!
+
+### !end-callout
+
+## Moving `isPresent` From `Student` to `App`
+
+Sofia's first goal is to create a single source of truth for her app. She wants to move all student data to one place.
+
+Sofia chooses to move the data to the `App` component. She chooses the `App` component because:
+
+1. It's more appropriate for the responsibility we gave `App`
+1. In React, we pass read-only data through `props`, which is passed into nested components. `Student` is more nested than `App`.
+
+### New Student Data Shape in `App`
+
+First, we should add `isPresent` into our student data in `App`.
+
+While we're here, we're also going to add `id` data to each student. This helps the data feel a little more realistic, as if we're reading it from a database or API.
+
+<!-- prettier-ignore-start -->
+```js
+function App() {
+  const studentData = [
+    {
+      id: 1,
+      nameData: 'Ada',
+      emailData: 'ada@dev.org',
+      isPresentData: false
+    },
+    {
+      id: 2,
+      nameData: 'Soo-ah',
+      emailData: 'sooah@dev.org',
+      isPresentData: false
+    },
+    {
+      id: 3,
+      nameData: 'Chrissy',
+      emailData: 'chrissy@dev.org',
+      isPresentData: true
+    }
+  ];
+
+  return (
+    <main>
+      <h1>Attendance</h1>
+      <StudentList students={studentData}></StudentList>
+    </main>
+  );
+}
+```
+<!-- prettier-ignore-end -->
+
+In this example, we've chosen the name `isPresentData` to help us clarify that this is coming from the student dataset. We can change the name to anything else if we want to!
+
+### Update PropTypes
+
+Now that `id` and `isPresentData` are in our `studentData`, and we pass that data into `StudentList` and `Student`, we should update our PropTypes.
```

Comment:
Sofia
provide an overview of what the change was and what the update needs to be
```suggestion
Since Sofia changed the structure of `studentData`, she also needs to update the PropTypes for the components that receive `studentData` through their `props`: `StudentList` and `Student`.

She added `id` and `isPresentData` to the `studentData`, so these need to be added to the PropTypes shape.
```